hw02/src/main/java/DIYarrayList.java

```diff
@@ -0,0 +1,243 @@
+import java.util.*;
+
+public class DIYarrayList<E> implements List<E> {
+
+    private int defaultCapacity = 10;
+    private int stepToIncCapacity;
+    private int size;
+    private Object[] elementData;
+
+    public DIYarrayList() {
+        this.elementData = new Object[defaultCapacity];
+    }
+
+    public DIYarrayList(int initialCapacity) {
+        this.elementData = new Object[initialCapacity];
+    }
+
+    public DIYarrayList(int initialCapacity, int defaultCapacity) {
+        this.elementData = new Object[initialCapacity];
+        this.stepToIncCapacity = defaultCapacity + 10;
+    }
```

Comment:
Все-равно немного странновато. Вы передаете параметр defaultCapacity, при этом у вас есть поле defaultCapacity. И этот параметр defaultCapacity никак не связан с полем defaultCapacity. 
И на основе параметра defaultCapacity вычисляете stepToIncCapacity .

Проще сразу передать stepToIncCapacity. 

```suggestion
    public DIYarrayList(int initialCapacity, int stepToIncCapacity) {
        this.elementData = new Object[initialCapacity];
        this.stepToIncCapacity = stepToIncCapacity;
    }
```

Или поясните ваше логику